DSpace/dspace-angular
# Search events bug-fix

## References
* Fixes #1811

## Description
This PR fixes a bug with search-events not being sent to the rest api on search pages.
This bug happens because our tracking-component uses `SearchConfigurationService`'s search options, which not every search component uses (e.g. the discover page), so it doesn't trigger a tracking event.

## Instructions for Reviewers
List of changes in this PR:
* Added `trackSearch()` to `SearchService`, using the same angularitics approach as the tracking component
* Modified `SearchComponent` to call this method whenever a search is performed and `trackStatistics` (a new input defaulting to false) is true
* Modified `SearchPageComponent`'s template to set `trackStatistics` to true and remove the previous tracking component
* The tracking component remains for usage on other places where it's still relevant

How to test:
* Visit the /search page
* Confirm a search event POST request is sent out to the rest api every time a search is performed
* Verify the same happens when a query is performed using the search input in the header

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
